Contribute/dotnet/dotnet-contribute.md

```diff
@@ -92,31 +110,23 @@ docs
             porting.vbproj
             porting-overview.vb
             Program.vb
+        /shared
+          /csharp ...
+          /fsharp ...
+          /vb ...
 ```
 
 > [!NOTE]
-> The language folders under snippets are not needed in the language guide area, where only one language is assumed.
+> The language folders under snippets are not needed in the language guide area, where only one language is assumed. For example, in the C# guide, it's assumed that all snippets are C#.
 
-The structure shown above includes one image, *portability_report.png*, and three code projects that include **code snippets** that are included in the *porting-overview.md* article. An accepted alternative structure contains one project per language that contains all snippets for all articles in that folder. This alternative has been used in the language reference areas because of very small snippets to demonstrate language syntax. It is discouraged in other areas.
+The structure shown above includes one image, *portability_report.png*, and three code projects that include **code snippets** that are included in the *porting-overview.md* article. 
 
-For historical reasons, many of the included snippets are stored under the */samples* folder in the *dotnet/docs* repository. If you're making major changes to an article, those snippets should be moved to the new structure. Do not move snippets for small changes.
+The *snippets/shared* folder is used for snippets that may span multiple articles within the same parent folder, such as the *porting* folder in the previous example. Only use the *shared* folder when you have a specific reason to do so, such as XAML code that's referenced by multiple articeles, yet can't compile in the *article-specific* folder.
```

Comment:
>The *snippets/shared* folder is used for snippets that may span multiple articles within the same parent folder,

What about snippets that span multiple articles under different parent folders?

Reply:
I would say that violates the basic principle of isolation. Do the shared C# snippets do that? I think they should break down and isolate further, otherwise it seems hard to manage.